test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageActionDescriptorProviderTest.cs

```diff
@@ -167,6 +168,41 @@ public void GetDescriptors_CopiesRouteValuesFromModel()
             Assert.Equal("Accounts/Test/{id:int?}", descriptor.AttributeRouteInfo.Template);
         }
 
+        [Fact]
+        public void GetDescriptors_CopiesActionConstraintsFromModel()
+        {
+            // Arrange
+            var expected = Mock.Of<IActionConstraint>();
+            var model = new PageRouteModel("/Areas/Accounts/Pages/Test.cshtml", "/Test", "Accounts")
+            {
+                Selectors =
+                {
+                    new SelectorModel
+                    {
+                        AttributeRouteModel = new AttributeRouteModel(),
+                        ActionConstraints = { expected }
+                    }
+                },
+            };
+            var applicationModelProvider = new TestPageRouteModelProvider(model);
+            var provider = new PageActionDescriptorProvider(
+                new[] { applicationModelProvider },
+                GetAccessor<MvcOptions>(),
+                GetRazorPagesOptions());
+            var context = new ActionDescriptorProviderContext();
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            var result = Assert.Single(context.Results);
+            var descriptor = Assert.IsType<PageActionDescriptor>(result);
+            Assert.Equal(model.RelativePath, descriptor.RelativePath);
+            Assert.Collection(
```

Comment:
Minor but it's weird to use `Assert.Collection(...)` for a 1-entry collection. Use `Assert.Same(expected, Assert.Single(descriptor.ActionConstraints))` or (if you prefer) split the assertions.